server/chat-plugins/random-battles.ts

```diff
@@ -281,11 +281,15 @@ export const commands: ChatCommands = {
 			return this.sendReplyBox(`<span style="color:#999999;">Moves for ${species.name} in ${formatName}:</span><br />${lgpeMoves}`);
 		}
 		if (!species.randomBattleMoves) {
-			return this.errorReply(`Error: No moves data found for ${species.name}${`gen${dex.gen}` in GEN_NAMES ? ` in ${GEN_NAMES[`gen${dex.gen}`]}` : ``}.`);
+			const gmaxSpecies = dex.getSpecies(`${args[0]}gmax`);
+			if (!gmaxSpecies.exists || !gmaxSpecies.randomBattleMoves) {
+				return this.errorReply(`Error: No moves data found for ${species.name}${`gen${dex.gen}` in GEN_NAMES ? ` in ${GEN_NAMES[`gen${dex.gen}`]}` : ``}.`);
+			}
+			species = gmaxSpecies;
 		}
 		const moves: string[] = [];
 		// Done because species.randomBattleMoves is readonly
-		for (const move of species.randomBattleMoves) {
+		for (const move of species.randomBattleMoves!) { // <- TypeScript bug: species.randomBattleMoves can't be undefined
```

Comment:
```suggestion
		for (const move of species.randomBattleMoves!) {
```

The `!` already means "this can be defined, but the mechanism by which it can be defined is too hard for TypeScript to check". I'm fine with comments that explain _why_ it's too hard for TS to check or something, but your comment as it stands doesn't seem to add anything.

Reply:
If you'd prefer not to have the assertion, you can also have a local `let randomBattleMoves = species.randomBattleMoves` variable.

Reply:
I added the comment because it seems like TypeScript should be able to tell that it’s defined, given the preceding checks, but OK.

Reply:
TypeScript is _sometimes_ nice enough to check properties, but it should never be relied upon.

Reply:
(Kris's fix is the one I suggested.)